int16[] start_x
int16[] start_y
int16[] start_d
int16[] goal_x
int16[] goal_y
int16[] goal_d
int16[] bot_num